shared/chat/conversation/info-panel/notifications/container.js

```diff
@@ -25,6 +25,19 @@ const serverStateToProps = (notifications: Types.NotificationsState, type: 'desk
   return 'never'
 }
 
+type StateProps = {
+  // Set only when the state has a selected conversation, inbox, and
+  // notifications object.
+  _props?: {
```

Comment:
having this `_props` seems a little redundant. I think you could do something like `StateProps = ?{...}` and then in `mergeProps` change `if (stateProps._props)` to `if (stateProps)`

Reply:
Sadly, I tried this and it didn't work. Looks like `mapStateToProps` is typed as:

```
  declare type MapStateToProps<S, OP: Object, SP: Object> = (
    state: S,
    ownProps: OP
  ) => ((state: S, ownProps: OP) => SP) | SP;
```

so `StateProps` has to be an object.

Reply:
we should not use this pattern. just make another component that checks and renderse nothing, else use this connected component. we don't want nested objects else your shall equals will always fail

Reply:
Ah, you're right. This is the component that checks and renders nothing, but I wanted to type the fact that all or none of the state props should be present, and the same for the merged props. I think I got another solution that avoids the nesting, PTAL!